src/rapidpro_abtesting/operations.py

```diff
@@ -924,6 +924,22 @@ def _get_flow_snippet(self, node, node_layout=None):
 }
 
 
+def count_occurrences(text, string):
+    if string[:6].lower() == "regex:":
+        pattern = string[6:]
+        return len(re.findall(pattern, string))
+    else:
+        return text.count(string)
+
+
+def replace_occurrences(text, string, replacement):
+    if string[:6].lower() == "regex:":
+        pattern = string[6:]
+        return re.sub(pattern, replacement, text)
+    else:
+        return text.replace(string, replacement)
```

Comment:
You could reduce the duplication by creating a function to return the regex pattern from a string, or `None`. Then, always try to extract the pattern, and use it if one exists or fall back.

I would recommend using `string.lower().startswith("regex:")` to test, and `string.replace("regex:", "")` to extract.

Recommend creating a constant for the string 'regex:', if it's used more than once.

I suggest removing '_occurrences' from the function names, leaving just 'count' and 'replace' - I don't think it will be in conflict with anything.

Reply:
Just realised that you'd have to use `string.replace("regex:", "", 1)`, or (new in Python 3.9) `string.removeprefix("regex:")`.

Reply:
Done, except that I stuck with slicing for prefix removal as that needs to be case insensitive.